### PR TITLE
Improved Github Action (Support PHP 5.6 => 8.1)

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Improved Github Action (Support PHP 5.6 => 8.1)
| Type?         | improvement
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Relative to PrestaShop/Prestashop#25921
| How to test?  | Github Actions are :green_circle:

### CODE REVIEW

File: tests/phpstan.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+docker rm -f temp-ps || true
+docker volume rm -f ps-volume || true
+
+docker run -tid --rm -v ps-volume:/var/www/html --name temp-ps prestashop/prestashop:$PS_VERSION
+
+# Clear previous instance of the module in the PrestaShop volume
+echo "Clear previous module"
+
+docker exec -t temp-ps rm -rf /var/www/html/modules/ps_customeraccountlinks
```
Comment (atomiix) [inline tests/phpstan.sh]:
```suggestion
docker exec -t temp-ps rm -rf /var/www/html/modules/ps_customersignin
```

File: tests/phpstan.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+# Run a container for PHPStan, having access to the module content and PrestaShop sources.
+# This tool is outside the composer.json because of the compatibility with PHP 5.6
+echo "Run PHPStan using phpstan-${PS_VERSION}.neon file"
+
+docker run --rm --volumes-from temp-ps \
+       -v $PWD:/var/www/html/modules/ps_customeraccountlinks \
```
Comment (atomiix) [inline tests/phpstan.sh]:
```suggestion
       -v $PWD:/var/www/html/modules/ps_customersignin \
```

File: tests/phpstan.sh [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+       -e _PS_ROOT_DIR_=/var/www/html \
+       --workdir=/var/www/html/modules/ps_customeraccountlinks phpstan/phpstan:0.12 \
```
Comment (atomiix) [inline tests/phpstan.sh]:
```suggestion
       --workdir=/var/www/html/modules/ps_customersignin phpstan/phpstan:0.12 \
```

File: tests/phpstan.sh [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+       analyse \
+       --configuration=/var/www/html/modules/ps_customeraccountlinks/tests/phpstan/phpstan-$PS_VERSION.neon
```
Comment (atomiix) [inline tests/phpstan.sh]:
```suggestion
       --configuration=/var/www/html/modules/ps_customersignin/tests/phpstan/phpstan-$PS_VERSION.neon
```

**Review by atomiix — CHANGES_REQUESTED** [review]

File: tests/phpstan.sh (hunk 4 above)
Comment (Progi1984) [reply tests/phpstan.sh]: Fixed

**Review by atomiix — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Pull request merged**